### ISSUE TITLE
When trie start supporting multiproof?

### ISSUE THREAD

**Issue by felicityin** [ethereumjs/ethereumjs-monorepo#2278]
A Merkle multiproof is a more efficient Merkle proof, for when it is necessary to prove the presence of a multiple elements that are in the same Merkle tree. Details: https://github.com/ledgerwatch/erigon/blob/devel/docs/programmers_guide/guide.md#multiproofs

When trie start supporting multiproof?

**Comment by jochem-brouwer** [comment]
We support verifying them (`verifyRangeProof`). Creating proofs is a WIP, see #2243.

**Comment by felicityin** [comment]
> We support verifying them (`verifyRangeProof`). Creating proofs is a WIP, see #2243.

Thanks! But I want t prove the presence of a multiple elements that do not constitute an interval. Maybe this will meet my needs better: https://github.com/ethereumjs/ethereumjs-monorepo/pull/2243/files#diff-31717aedba209c646988550ff9a91ee518d515d3693d568651dd28f7681c0683. But it has not been merged yet.

**Comment by jochem-brouwer** [comment]
I assumed your erigon page was a page about multiproofs for intervals, but apparently there is also a way to support multiproofs for non-intervals? I will have to read the docs about this to see if this is actually possible. But in that case, this is not being worked on yet.

For the time being, would it also work to just batch `createProof` and then use this to provide proofs for multiple elements (this of course has duplicate elements)?

**Comment by felicityin** [comment]
> I assumed your erigon page was a page about multiproofs for intervals, but apparently there is also a way to support multiproofs for non-intervals? I will have to read the docs about this to see if this is actually possible. But in that case, this is not being worked on yet.
> 
> For the time being, would it also work to just batch `createProof` and then use this to provide proofs for multiple elements (this of course has duplicate elements)?

There is a way to support multiproofs for non-intervals in Sparse Merkle: https://arxiv.org/pdf/2002.07648.pdf. It is really useful.

**Comment by jochem-brouwer** [comment]
Very interesting, this sounds fun to implement!

**Comment by felicityin** [comment]
> Very interesting, this sounds fun to implement!

Yes, it is.

**Comment by ScottyPoi** [comment]
@jochem-brouwer 
This is different than the `RangeProof` work you're doing, right?
If so I will start looking into it.

**Comment by jochem-brouwer** [comment]
Yes: my current PR (i.e https://github.com/ethereumjs/ethereumjs-monorepo/pull/2243) is about sequential proofs. This paper shows that it is also possible to prove for non-sequential proofs, so this is indeed diffferent.
